src/common/models/Projection.ts

```diff
@@ -454,8 +472,9 @@ export class Projection {
    * to reporting weirdness), we skip that data point and may average fewer
    * than 7 days worth of data.
    */
-  getWeeklyAverageCaseForDay(i?: number) {
-    const lastIndex = this.indexOfLastValue(this.dailyPositiveTests);
+
+  getWeeklyAverageForDay(data: Array<number | null>, i?: number) {
+    const lastIndex = this.indexOfLastValue(data);
```

Comment:
Isn't this the same as `smoothWithRollingAverage`? I think that the only difference is that `smoothWithRollingAverage` sets `null` for the first 6 days, whereas this one calculates the rolling average with a smaller window.

Reply:
Yeah...   It's the same concept but implemented differently.  I think we should consolidate and use smoothWithRollingAverage(), but I don't feel strongly it's worth reworking things in this PR or not. 😁 

TBH this whole class has turned into a bit of a monster.  I think it'd be worth a pretty major overhaul, perhaps factoring out all of the timeseries-related code into a Timeseries class or something.

NOTE: One thing I was worried about is that this code has handling for negative values but smoothWithRollingAverage() does not...  But I _think_ we actually ended up solving negative values in `smoothCumulatives()`, so it probably doesn't matter.

Reply:
punting this for now as well